From: Mathias Gumz <akira at fluxbox dot org>
Date: Wed, 15 Oct 2008 21:31:37 +0000 (+0200)
Subject: additional fix for the resize bug
X-Git-Url: http://git.fluxbox.org/?p=fluxbox.git;a=commitdiff_plain;h=6642792f68ae42ba8d2bcbe2033284f7e8e61167

additional fix for the resize bug
---

From: Peter Hercek <hercek at sf dot net>
Date: Sun, 11 Jan 2009 13:23:08 +0000 (+0100)
Subject: fixed resize bug (caused by typo, see #2498507)
X-Git-Url: http://git.fluxbox.org/?p=fluxbox.git;a=commitdiff_plain;h=4e831484d55398e5c328aeb531060ffaf997f300

fixed resize bug (caused by typo, see #2498507)
---

diff --git a/src/FbWinFrame.cc b/src/FbWinFrame.cc
index ce25004..711e993 100644
--- a/src/FbWinFrame.cc
+++ src/FbWinFrame.cc
@@ -38,6 +38,7 @@
 #include <algorithm>
 #include <X11/X.h>
 
+using std::max;
 using std::mem_fun;
 using std::string;
 
@@ -1702,7 +1703,8 @@ int FbWinFrame::yOffset() const {
 
 void FbWinFrame::applySizeHints(unsigned int &width, unsigned int &height,
                                 bool maximizing) const {
-    height -= titlebarHeight() + handleHeight();
+    const int h = height - titlebarHeight() - handleHeight();
+    height = max(h, static_cast<int>(titlebarHeight() + handleHeight()));
     sizeHints().apply(width, height, maximizing);
     height += titlebarHeight() + handleHeight();
 }
